### PR TITLE
Adjust Labels and Titles

### PR DESCRIPTION
**Pull request by philippfromme**
Adjusts labels and titles according to https://docs.google.com/document/d/1zuXLaRhNXdLa3Q4CV9J39-H5b22jDDwXotzfQV1ZsJw/edit#heading=h.bcw00wds71dr.

image

### CODE REVIEW

**Comment by nikku** [comment]
Are we sure we want to do this? It is (1) a major breaking change regarding i18n and (2) does not contribute to make these menus more readable (i think).

**Comment by nikku** [comment]
Good thing, of course, is that it'll now be somewhat consistent across our code-base.

**Comment by barmac** [comment]
If we are to merge this, we should update the translations in https://github.com/bpmn-io/bpmn-js-i18n accordingly.

**Comment by barmac** [comment]
Note that Polish translations already use sentence case: https://github.com/bpmn-io/bpmn-js-i18n/blob/main/translations/pl.js

**Comment by barmac** [comment]
So changes would be:
```diff
-    "Add Lane above": "Dodaj pas powyżej",
+    "Add lane above": "Dodaj pas powyżej",
```

**Comment by barmac** [comment]
BTW this PR shows that `translate('Text in English')` pattern, while convenient, can lead us into troubles. Any change in the labels is in fact breaking for the translations. What we could do instead is to use static labels to be replaced with English or other language variant in runtime, e.g. 
```javascript
// in code
translate('BOUNDARY_EVENT')

// en.json
{
  "BOUNDARY_EVENT": "Boundary event"
}
```

**Comment by nikku** [comment]
@barmac This change will be reflected in a single commit, in the i18n repository, I hope. It will also be a breaking change for the library so integrators have to expect breaking things. 

I don't think that it will be much of a hazzle for our contributors to update, if we provide guidance :arrow_right: Blog is the minimum to ensure they have a chance. Even better updating the translation keys (we could have an old to new English translation).

@philippfromme Please create a tracking issue to collect this and @barmac's thoughts, the repositories you touch and let's outline a roll-out strategy before merging this PR. It is a huge change (for some) and we want to handle it appropriately, with a little bit of planning.

**Comment by philippfromme** [comment]
@nikku @barmac Please have a look at https://github.com/bpmn-io/internal-docs/issues/861 and let me know what you think. The refactoring proposed by @barmac could be a follow-up or done along with the adjustment of the labels.

File: docs/translations.json
Comment (barmac) [inline docs/translations.json:111]: In the docs, it's treated as a single word: https://docs.camunda.io/docs/components/modeler/bpmn/subprocesses/
While in the specification, it takes form of "Sub-Process". I think the hyphened version is correct here.

**Review by barmac — APPROVED** [review]

**Review by nikku — CHANGES_REQUESTED** [review]
Let's clean up the commits.

Let's clearly mark this as a breaking change, so we don't forget the major version bump.

Let's turn this from `chore` to `feat`, as we believe (I believe) this is a significant improvement to the existing editor. We'd otherwise not ship it.

**Pull request merged**
